torchrl/envs/common.py

```diff
@@ -220,8 +220,18 @@ def __init__(
     @classmethod
     def __new__(cls, *args, **kwargs):
         cls._inplace_update = True
+        if not hasattr(cls, "_batch_locked"):
```

Comment:
Why can't we do this
```
    @classmethod
    def __new__(cls, *args, _batch_locked=True, **kwargs):
        cls. _batch_locked = _batch_locked
```
and then for all subclasses
```
    @classmethod
    def __new__(cls, *args, **kwargs):
        super().__new__(_batch_locked = what_you_want_it_to_be)
```

Perhaps you have tried but it does not work? Curious about this

Reply:
Haven't tried this way, checking for the cls to have the arge felt more intuitive for me since it's the place the parameters are set. It's also more robust to changes in the default EnvBase. However if you prefer to have it like above tell me and i will make the changes